sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/AppendBlobClient.java

```diff
@@ -159,24 +161,14 @@ public AppendBlobItem appendBlock(InputStream data, long length) {
      * @param context Additional context that is passed through the Http pipeline during the service call.
      *
      * @return A {@link Response} whose {@link Response#value() value} contains the append blob operation.
+     * @throws UnexpectedLengthException when the length of data does not match the input {@code length}.
```

Comment:
You cannot throw UnexpectedLengthException here as it is not public API. You either need to make it public API, or not throw it from public API.

Reply:
Put the exception in public exception folder